### DISCUSSION TITLE
Is it possible to remember the sort order for a directory?

### DISCUSSION THREAD

**Discussion by a-torgovitsky** [sxyazi/yazi#2131] [1. Q&A] [answered]
### What system are you running Yazi on?

Linux X11

### What terminal are you running Yazi in?

kitty 36.2

### `yazi --debug` output

```Shell
N/A
```

### Describe the question

My default sort order is by last modified. Sometimes, I go into a directory `dir1` and change the sort order to name ascending. If I go into a subdirectory `sdir2` and back to `dir1`, the sort order has reset to the default of last modified.

I think this is expected behavior for yazi, but I'm wondering if it's possible to modify yazi to remember sort orders? Some GUI file browsers have this type of behavior, but I suppose it requires keeping some sort of memory of folders visited and last sort order, so perhaps it's outside of the scope of the yazi project.

### Checklist

- [X] I have searched the existing discussions/issues

**Comment by martinengelke** [comment]
Maybe we could provide a dotfile in those dirs like `.yzdsrt` or something similar which has some metadata about the sorting order and Yazi can check those files when it enters a folder?

**Reply by a-torgovitsky** [reply]
If that were the solution, then I suppose one would also want to save toggle hidden status, linemode...others?

**Reply by martinengelke** [reply]
Uhhh and custom sorting like I already asked for :)

**Reply by a-torgovitsky** [reply]
Could also be used to "pin" directories at the top or color-code/change font of certain directories so that they are more visible

**Reply by jheroy** [reply]
Additionally, the sorting in the folder preview panel should remain consistent with the sorting after entering the folder.

**Comment by AnirudhG07** [comment]
Hi @a-torgovitsky Check out https://yazi-rs.github.io/docs/tips#folder-rules & https://yazi-rs.github.io/docs/tips#folder-previewer
Here you can setup folder specific sorting, previewers, etc. 
Hope this solves your issues!

**Reply by a-torgovitsky** [reply]
I'm not sure I see how those can be used to create this functionality? I actually already use folder-rules to have specific folders sorted in a certain order by default (like `Downloads`, as shown in the example). This question was more about being able to save/update that default dynamically. If you see a way to do this with folder-rules, could you explain please?

**Comment by boydaihungst** [comment, marked as answer]
https://github.com/boydaihungst/pref-by-location.yazi

Try this plugins. It's conflict with folder-rules.

**Reply by a-torgovitsky** [reply]
Your plugin works perfectly and does exactly what I want. Thanks for writing it!
